sql/src/main/java/io/crate/execution/ddl/tables/AlterTableRequest.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.execution.ddl.tables;
+
+import io.crate.metadata.RelationName;
+import org.elasticsearch.ElasticsearchGenerationException;
+import org.elasticsearch.action.support.master.AcknowledgedRequest;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.bytes.BytesArray;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+
+import static org.elasticsearch.common.settings.Settings.readSettingsFromStream;
+import static org.elasticsearch.common.settings.Settings.writeSettingsToStream;
+
+public class AlterTableRequest extends AcknowledgedRequest<AlterTableRequest> {
+
+    private final RelationName tableIdent;
+    @Nullable
+    private final String partitionIndexName;
+    private final boolean isPartitioned;
+    private final boolean excludePartitions;
+    private final Settings settings;
+    @Nullable
+    private final String source;
+
+    public AlterTableRequest(RelationName tableIdent,
+                             @Nullable String partitionIndexName,
+                             boolean isPartitioned,
+                             boolean excludePartitions,
+                             Settings settings,
+                             Map mappingSource) {
```

Comment:
Not sure I get why we call this `(mapping)Source` ? Isn't this the mappingDelta / diff we want to be applied?

Reply:
Is this comment only related to the constructor name argument or to the whole property of the class ?